react-state/my-app/src/ImageCaption.tsx

```diff
@@ -0,0 +1,14 @@
+import { useState } from 'react';
+
+type Props = {
+  imageCaps: string[];
+};
+
+export function ImageCaption({ imageCaps }: Props) {
+  const [state, setState] = useState(0);
+  return (
+    <div>
+      <h3 onClick={() => setState(state + 1)}>{imageCaps[state % 3]}</h3>
```

Comment:
Nice, but it would be nicer to use imageCaps.length instead of 3 so that your code still works even if you pass it more captions.